engine/src/flutter/lib/ui/window.dart

```diff
@@ -968,6 +969,20 @@ class AccessibilityFeatures {
   /// Only supported on iOS.
   bool get onOffSwitchLabels => _kOnOffSwitchLabelsIndex & _index != 0;
 
+  /// Whether accessibility announcements (like [SemanticsService.announce])
+  /// are allowed on the current platform.
+  ///
+  /// Returns `false` on Android, where platform announcements are deprecated
```

Comment:
Remove "on android" because the point of this method is to abstract away the platform. 

Consider additional modifications based on this feedback below.